MobileWallet/Screens/Home/Transaction/TransactionViewController.swift

```diff
@@ -481,3 +442,51 @@ class TransactionViewController: UIViewController, UITextFieldDelegate {
         }
     }
 }
+
+extension TransactionViewController: UITextFieldDelegate {
+    func textFieldShouldBeginEditing(_ textField: UITextField) -> Bool {
+        return isEditingContactName
+    }
+
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        guard textField.text?.isEmpty == false else {
+            textField.text = contactAlias
+            return false
+        }
+
+        isEditingContactName = false
+
+        guard contactPublicKey != nil else {
+            UserFeedback.shared.error(
+                title: NSLocalizedString("Contact error", comment: "Transaction detail screen"),
+                description: NSLocalizedString("Missing public key from transaction.", comment: "Transaction detail screen")
+            )
+            return true
+        }
+
+        do {
+            try TariLib.shared.tariWallet!.addUpdateContact(alias: textField.text!, publicKey: contactPublicKey!)
+
+            DispatchQueue.main.asyncAfter(deadline: .now() + 0.25, execute: {
+                UINotificationFeedbackGenerator().notificationOccurred(.success)
+                //UserFeedback.shared.success(title: NSLocalizedString("Contact Updated!", comment: "Transaction detail screen"))
+            })
+
+        } catch {
+            UserFeedback.shared.error(
+                title: NSLocalizedString("Contact error", comment: "Transaction detail screen"),
+                description: NSLocalizedString("Failed to save contact details.", comment: "Transaction detail screen"),
+                error: error
+            )
+        }
+
+        return true
+    }
+
+    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
+        let maxLength = 40
```

Comment:
Why set this in 2 places?

Reply:
i seen the same issue on another controller, while editing name on Transaction Detail Controller